File: python/agentbay/models/mcp_tool.py
from dataclasses import dataclass
from typing import Dict, Any


@dataclass
class McpTool:
    """
    Represents an MCP tool with complete information.
    """
    name: str
    description: str
    input_schema: Dict[str, Any]
    server: str
    tool: str

    def get_name(self) -> str:
        """Return the tool name."""
        return self.name

    def get_server(self) -> str:
        """Return the server name that provides this tool."""
        return self.server 

File: java/agentbay/src/main/java/com/aliyun/agentbay/mcp/McpTool.java
package com.aliyun.agentbay.mcp;

import java.util.Map;

public class McpTool {
    private String name;
    private String description;
    private Map<String, Object> inputSchema;
    private String server;
    private String tool;

    public McpTool() {
    }

    public McpTool(String name, String description) {
        this.name = name;
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Map<String, Object> getInputSchema() {
        return inputSchema;
    }

    public void setInputSchema(Map<String, Object> inputSchema) {
        this.inputSchema = inputSchema;
    }

    public String getServer() {
        return server;
    }

    public void setServer(String server) {
        this.server = server;
    }

    public String getTool() {
        return tool;
    }

    public void setTool(String tool) {
        this.tool = tool;
    }
}
